cmd/snapshot-controller/main.go

```diff
@@ -140,23 +143,15 @@ func main() {
 	// Create and register metrics manager
 	metricsManager := metrics.NewMetricsManager()
 	wg := &sync.WaitGroup{}
-	wg.Add(1)
+
+	mux := http.NewServeMux()
 	if *httpEndpoint != "" {
-		srv, err := metricsManager.StartMetricsEndpoint(*metricsPath, *httpEndpoint, promklog{}, wg)
+		err := metricsManager.PrepareMetricsEndpoint(mux, *metricsPath, promklog{})
```

Comment:
Should this be called "PrepareMetricsPath" as the httpEndpoint is not used yet?

Reply:
yeah, will update this